Tests/ApolloCodegenTests/URLDownloaderTests.swift

```diff
@@ -0,0 +1,131 @@
+@testable import ApolloCodegenLib
+import ApolloTestSupport
+import ApolloCodegenTestSupport
+import XCTest
+
+class URLDownloaderTests: XCTestCase {
+  let urlRequest = URLRequest(url: TestURL.mockServer.url)
+  let downloadURL = URL(string: "file://anywhere/nowhere/somewhere")!
+  let defaultTimeout = 0.5
+  var downloader: URLDownloader!
+  var session: MockNetworkSession!
+
+  override func setUp() {
+    downloader = URLDownloader()
+    session = nil
+  }
+
+  override func tearDown() {
+    downloader = nil
+    session = nil
+  }
+
+  private func setRequestHandler(statusCode: Int, data: Data? = nil, error: Error? = nil, abandon: Bool = false) {
```

Comment:
The unfortunate side-effect is that if you do not call `setRequestHandler` in one of the tests it'll fail with an obscure `Could not connect to the server.` error because `URLDownloader` defaults to using the `URLSession` extension for `NetworkService`. Ideally it would tell you that you had not set a request handler but I couldn't find a great way to do that.

Reply:
I would throw a comment on this method explaining that it needs to be called in each of the test methods then

Reply:
A bit more thinking and I found the solution I was looking for - 3d7efb146. If you forget you'll now get this error message to guide you, `failed - You must call setRequestHandler before using downloader!`.